pkg/status/workstatus.go

```diff
@@ -83,25 +83,37 @@ func (c *Controller) syncSingletonWorkStatus(ctx context.Context, ref singletonW
 	return c.syncWorkStatus(ctx, workStatusRef(ref))
 }
 
-func updateObjectStatus(ctx context.Context, objectIdentifier *util.ObjectIdentifier, status map[string]interface{},
+func updateObjectStatus(ctx context.Context, objectIdentifier util.ObjectIdentifier, status map[string]interface{},
 	listers util.ConcurrentMap[schema.GroupVersionResource, cache.GenericLister], wdsDynClient dynamic.Interface) error {
 	logger := klog.FromContext(ctx)
 
 	gvr := objectIdentifier.GVR()
 	lister, found := listers.Get(gvr)
 	if !found {
-		logger.V(2).Info("Could not find lister for gvr", "gvr", gvr)
+		logger.V(4).Info("Could not find lister for gvr", "gvr", gvr)
 		return nil
 	}
 
-	obj, err := lister.ByNamespace(objectIdentifier.ObjectName.Namespace).Get(objectIdentifier.ObjectName.Name)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			logger.V(2).Info("Could not find object", "objectIdentifier", objectIdentifier)
-			return nil
+	var obj runtime.Object
+	var err error
+	if objectIdentifier.ObjectName.Namespace == "" {
+		obj, err = lister.Get(objectIdentifier.ObjectName.Name)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				logger.V(4).Info("Could not find object", "objectIdentifier", objectIdentifier)
```

Comment:
I would make this log message a little more explanatory, something like "Abandoning update for non-existent workload object" or "Did not update workload object because it is not in local cache, presumably because it was recently deleted".